packages/lsp/src/connection_manager.ts

```diff
@@ -42,6 +42,13 @@ export class DocumentConnectionManager
     Private.setLanguageServerManager(options.languageServerManager);
   }
 
+  /**
+   * The path of the active widget adapter.
+   */
+  get currentAdapter(): string | null {
+    return this._currentAdapter;
+  }
```

Comment:
The current adapter contains the path to the active adapter instead of the adapter itself, because when adding a new adapter,  `app.shell.currentChanged` is emitted before the adapter is registered.